File: frontend/frontend/src/types/issue.ts
export const IssuePriority = {
  LOW: 0,
  MEDIUM: 1,
  HIGH: 2,
} as const;

export type IssuePriority =
  typeof IssuePriority[keyof typeof IssuePriority];

export const IssueStatus = {
  OPEN: 0,
  IN_PROGRESS: 1,
  CLOSED: 2,
} as const;

export type IssueStatus =
  typeof IssueStatus[keyof typeof IssueStatus];

export interface Attachment {
  id: string;
  fileName: string;
  fileUrl: string;
}

export interface IssueAssignee {
  id: string;
  name: string;
  email?: string;
}

export interface Issue {
  id: string;
  title: string;
  description?: string;
  priority: IssuePriority;
  status: IssueStatus;
  assigneeId?: string;
  assignee?: IssueAssignee;
  attachments: Attachment[];
}

export interface CreateIssueRequest {
  title: string;
  description: string;
  priority: IssuePriority;
}

export interface UpdateIssueRequest {
  status: IssueStatus;
  priority: IssuePriority;
  assigneeId?: string | null;
}

File: backend/IssueTracker/IssueTracker.Domain/Entities/Issue.cs
using IssueTracker.Domain.Enums;

namespace IssueTracker.Domain.Entities;

public class Issue
{
    public Guid Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public IssueStatus Status { get; set; } = IssueStatus.Open;

    public IssuePriority Priority { get; set; } = IssuePriority.Low;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public Guid? AssigneeId { get; set; }

    public User? Assignee { get; set; }

    public List<Attachment> Attachments { get; set; } = [];
}